### PR TITLE
8286172: Create an automated test for JDK-4516019

### PR DESCRIPTION
**Pull request by srmandal**
Create an automated test for [JDK-4516019](https://bugs.openjdk.java.net/browse/JDK-4516019)

Clicking on the increment/decrement buttons of the spinner does not install focus on the spinner or one of its focusable children.

The test validates the same.

This review is for migrating tests from a closed test suite to open.

Testing:
The test ran successfully on Mach5 with multiple runs (30) on windows-x64, linux-x64 and macos-x64.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8286172](https://bugs.openjdk.org/browse/JDK-8286172): Create an automated test for JDK-4516019

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Alisen Chung](https://openjdk.org/census#achung) (@alisenchung - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/8546/head:pull/8546` \
`$ git checkout pull/8546`

Update a local copy of the PR: \
`$ git checkout pull/8546` \
`$ git pull https://git.openjdk.org/jdk pull/8546/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8546`

View PR using the GUI difftool: \
`$ git pr show -t 8546`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/8546.diff">https://git.openjdk.org/jdk/pull/8546.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/swing/4516019/JSpinnerFocusTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            Thread.sleep(300);
+            // Move cursor to click spinner up arrow button
+            robot.mouseMove(bounds.x + bounds.width - 2,
+                bounds.y + bounds.height / 4);
+            robot.delay(300);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.delay(300);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (prrace) [inline test/jdk/javax/swing/4516019/JSpinnerFocusTest.java:113]: Shouldn't the test fail if we get an exception ?

**Review by prrace — CHANGES_REQUESTED** [review]

**Review by prrace — CHANGES_REQUESTED** [review]
Also you are doing the same thing I've told others not to - look at the path name

test/jdk/javax/swing/4516019/JSpinnerFocusTest.java

what is the point of 4516019 ? Do NOT create an extra directory for no reason.
especially one just numbered for a bugid
and in this case this test should go in the pre-existing JSpinner directory !

File: test/jdk/javax/swing/4516019/JSpinnerFocusTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 4516019
+ * @summary Clicking on the increment/decrement buttons of the spinner
+ * does not get focus onto the spinner.
```
[the remark this answers is not in the document]
Comment (srmandal) [reply test/jdk/javax/swing/4516019/JSpinnerFocusTest.java:29]: Changed appropriately.

File: test/jdk/javax/swing/4516019/JSpinnerFocusTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    private void runTest() {
+        try {
+            Robot robot = new Robot();
+            Rectangle bounds = new Rectangle(jSpinner.getLocationOnScreen(),
+                jSpinner.getSize());
+
+            // Move cursor to place it in the spinner editor
+            robot.mouseMove(bounds.x + bounds.width/2 ,
+                bounds.y + bounds.height /2);
+            robot.delay(300);
```
[the remark this answers is not in the document]
Comment (srmandal) [reply test/jdk/javax/swing/4516019/JSpinnerFocusTest.java:100]: Done

File: test/jdk/javax/swing/4516019/JSpinnerFocusTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            Thread.sleep(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                jTextFieldFocusStatus = ((DefaultEditor) jSpinner.getEditor())
+                    .getTextField().isFocusOwner();
+            });
+            if (!jTextFieldFocusStatus) {
+                throw new RuntimeException(
+                    "Clicking on JSpinner buttons did not"
+                        + " shift foucs to the JSpinner");
+            }
```
[the remark this answers is not in the document]
Comment (srmandal) [reply test/jdk/javax/swing/4516019/JSpinnerFocusTest.java:81]: Done

**Comment by srmandal** [comment]
> Also you are doing the same thing I've told others not to - look at the path name
> 
> test/jdk/javax/swing/4516019/JSpinnerFocusTest.java
> 
> what is the point of 4516019 ? Do NOT create an extra directory for no reason. especially one just numbered for a bugid and in this case this test should go in the pre-existing JSpinner directory !

As suggested:
Test removed from the original location. 
Test added under pre-existing JSpinner directory.

File: test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        jFrame.add(jSpinner, BorderLayout.CENTER);
+        jFrame.setLocationRelativeTo(null);
+        jFrame.setSize(300, 300);
+        jFrame.setVisible(true);
+    }
+
+    public void doTest() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            Thread.sleep(1000);
```
Comment (alisenchung) [inline test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java:69]: Would it be better to make robot a global variable and use Robot.waitForIdle() here? Sleeping might slow down the test unnecessarily

File: test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            runTest();
+            Thread.sleep(1000);
```
Comment (alisenchung) [inline test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java:71]: same here

**Review by alisenchung — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java (hunk 6 above)
Comment (srmandal) [reply test/jdk/javax/swing/JSpinner/JSpinnerFocusTest.java]: Fixed as suggested.

**Review by alisenchung — APPROVED** [review]

**Comment by srmandal** [comment]
Hi! Can someone please take a look at this PR and provide their valuable feedback and approval?

**Comment by srmandal** [comment]
Hi! Can someone please take a look at this PR and provide their valuable feedback and approval?

**Comment by srmandal** [comment]
Hi! Can someone please sponsor this PR?

**Review by prrace — APPROVED** [review]

**Comment by srmandal** [comment]
Can someone sponsor this PR?

**Pull request closed without merging**